### PR TITLE
animation-iteration-count property is a number instead of integer.

### PR DESCRIPTION
**Pull request by hiikezoe**
animation-iteration-count is defined as 'infinite | &lt;number&gt;'.

https://www.w3.org/TR/css3-animations/#animation-iteration-count-property
---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors

- [X] There are tests for these changes
Though I don't know whether the test case runs or not, there is a test case.
https://github.com/servo/servo/blob/42ec8a89747ba1e25769fffae083f62fe9576987/tests/wpt/css-tests/css-animations-1_dev/xhtml1/animation-iteration-count-004.xht#L17

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14732)

### CODE REVIEW

File: components/style/properties/longhand/box.mako.rs [hunk 1]
```diff
@@ -753,12 +753,12 @@ ${helpers.single_keyword("overflow-x", "visible hidden scroll auto",
...
 
-                let number = try!(input.expect_integer());
-                if number < 0 {
+                let number = try!(input.expect_number());
+                if number < 0.0 {
                     return Err(());
                 }
 
-                Ok(AnimationIterationCount::Number(number as u32))
+                Ok(AnimationIterationCount::Number(number as f32))
```
Comment (Wafflespeanut) [inline components/style/properties/longhand/box.mako.rs]: This cast won''t be necessary. It's already parsed as an `f32`.

**Comment by Wafflespeanut** [comment]
Yes, we usually have parse tests for this. But, I don't see any for `animation-*` (@emilio Do we need one?)

File: components/style/properties/longhand/box.mako.rs [hunk 2]
```diff
@@ -743,7 +743,7 @@ ${helpers.single_keyword("overflow-x", "visible hidden scroll auto",
         #[derive(Debug, Clone, PartialEq)]
         #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
         pub enum AnimationIterationCount {
```
Comment (emilio) [inline components/style/properties/longhand/box.mako.rs]: Let's also add a link to https://drafts.csswg.org/css-animations/#animation-iteration-count here, or on top of the parse function.

**Review by emilio — COMMENTED** [review]
Yeah, let's add a test for this property. A WPT test would be top, though a unit test can be enough I guess.

r=me, with that and the unneeded cast removed.

**Comment by emilio** [comment]
Oh, and also, thanks for the patch again @hiikezoe! :)

**Comment by hiikezoe** [comment]
@emilio thank you for reviewing!  I am going to add a unit test for this but I don't know where it should be.  Could you please tell me where it should be?

**Comment by Wafflespeanut** [comment]
`tests/unit/style/parsing/`

You can find a number of parse tests for other properties there. Maybe we should create a new file `animation.rs` and put our test there.

**Comment by emilio** [comment]
The last commit looks fine to me, but it seems like you pushed a bunch of extra commits that shouldn't be part of this PR.

**Comment by KiChjang** [comment]
@hiikezoe I highly recommend you to consult this guide if you missed it: https://github.com/servo/servo/wiki/Github-workflow.

**Comment by hiikezoe** [comment]
I did send a new PR. I am sorry for confusion.

**Pull request closed without merging**